### DISCUSSION TITLE
Appreciation and Suggestion for Course Improvement

### DISCUSSION THREAD

**Discussion by ArnaudBand** [Cyfrin/foundry-full-course-cu#1508] [Ideas]
Hi @PatrickAlphaC,

I'm really enjoying the [Cyfrin](https://updraft.cyfrin.io/) course and wanted to say thank you for putting it together. The content is clear and informative.

I have a suggestion that I think could make the course even better. It would be great if we could have some exercises or tasks after each lesson. These could help reinforce what we've learned and make the course more interactive.

Thanks again for the great course!

**Comment by EngrPips** [comment]
Hello @ArnaudBand , Glad to hear you are enjoying the course and you learning a ton. About your suggestion of having some exercises or task after each lesson i think there is some sort of exercise to mint NFT after each lesson or you think those aren't enough of a task.

**Reply by ArnaudBand** [reply]
Hi @EngrPips, Thank you for the message. However, I'm on lesson 9 but didn't see any exercises. Can you please send the link for the exercises or where I can access them?

**Reply by ARomano79** [reply]
Have you noticed any QR codes pop on the screen for the last minute or so of certain videos?

**Reply by EngrPips** [reply]
@ARomano79 , does this happen both on updraft and youtube ?

**Reply by ARomano79** [reply]
Just on YouTube, it pops up for the last minute or so on the section end/congratulatory videos.

**Reply by EngrPips** [reply]
Oh i see, thanks for the heads up

**Comment by EngrPips** [comment]
Hello @ArnaudBand , You should check this contract address `0x76B50696B8EFFCA6Ee6Da7F6471110F334536321` on sepolia etherscan as it is the main contract for the foundry-full-course challenge, you should visit the read section of the contract as shown in the picture below 

<img width="1440" alt="Screenshot 2024-03-21 at 1 07 44 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/eeab9211-e154-43e0-97a2-a28be2d5045a">

You should see the `getChallengeContract` getter function, for that function you can enter the section number for which you want to get it `challengeContract`, when you get the address you can look it up on sepolia etherscan and interact with the contract to mint the NFT for that section. If anything is confusing please don't hesitate to ask

**Comment by PatrickAlphaC** [comment]
This is a good shout! We've been meaning to add quizzes. The challenge NFTs are nice, but they are really small. 

What kind of challenges would you like to see?

**Reply by ArnaudBand** [reply]
Thanks for your response. I was meaning, it could more better to test our knowledge after each lessons. That we will boost our knowledge.

**Reply by EngrPips** [reply]
Indeed !

**Comment by noretake** [comment]
> Hi @PatrickAlphaC,
> 
> I'm really enjoying the [Cyfrin](https://updraft.cyfrin.io/) course and wanted to say thank you for putting it together. The content is clear and informative.
> 
> I have a suggestion that I think could make the course even better. It would be great if we could have some exercises or tasks after each lesson. These could help reinforce what we've learned and make the course more interactive.
> 
> Thanks again for the great course!

I am getting locked in... Grateful to have met your path and rails here. Let me lock in as well.....

**Comment by Elijah-ug** [comment]
What did I miss... Inspired to get started and take it through all the way... through the smart contracts fire...
